Example/ViewController.swift

```diff
@@ -541,3 +573,43 @@ extension ViewController: VisualInstructionDelegate {
         return presented
     }
 }
+
+// MARK: Free driving
+extension ViewController {
+    func trackLocations(mapView: NavigationMapView) {
+        let dataSource = PassiveLocationDataSource(directions: Settings.directions)
+        let locationManager = PassiveLocationManager(dataSource: dataSource)
+        mapView.locationManager = locationManager
```

Comment:
This is the bare minimum necessary to integrate free-driving localization into an application’s pre-navigation map view. The rest of the changes in this file are related to a nifty but inessential feature that plots the user’s raw and idealized path on the map.